File: flare-web-client/src/flareclient-extensions-builtin/tasks/haptic/haptic-task.js
export const HapticTask = {
  id: "haptic",
  execute(params) {
    if (!navigator.vibrate) return;
    const style = (params && params.style) || "success";
    const patterns = {
      light: 30, medium: 60, heavy: 100, success: 50,
      warning: [40, 80, 40], error: [40, 60, 40, 60, 80]
    };
    navigator.vibrate(patterns[style] || patterns.success);
  }
};

File: flare-android-client/flareclient-extensions-builtin/src/main/java/dev/flareframework/extensions/builtin/tasks/haptic/HapticTask.java
package dev.flareframework.extensions.builtin.tasks.haptic;

import android.app.Activity;
import dev.flareframework.client.FlareClientActivity;
import dev.flareframework.client.flare.task.FlareClientTask;
import org.json.JSONObject;

/**
 * Delegates to FlareClientActivity.triggerHaptic(style), which already
 * implements the full success/warning/error/light/medium/heavy pattern
 * set. Do not reimplement with HapticFeedbackConstants here.
 */
public class HapticTask implements FlareClientTask {
    public static final String ID = "haptic";

    @Override
    public String id() { return ID; }

    @Override
    public void execute(Activity host, JSONObject params) {
        if (host instanceof FlareClientActivity) {
            ((FlareClientActivity) host).triggerHaptic(params.optString("style", "success"));
        }
    }
}